tests/test_input_manager.py

```diff
@@ -1981,6 +1981,32 @@ def test_add_variable_to_pool_valid(
     mocker.patch.object(input_manager, "_InputManager__metadata", mock_metadata_for_add_variable_to_pool)
     mocker.patch.object(input_manager, "_InputManager__pool", starting_im_pool)
     mocker.patch.object(input_manager, "_validate_input_by_type", return_value=True)
+    # Capture the original method
+    original_add_to_pool = input_manager._add_to_pool
+    original_check_modifiability = input_manager._check_modifiability
+    original_validate_data = input_manager._validate_data
+    orginal_prepare_data = input_manager._prepare_data
+
+    def side_effect_add_pool(*args, **kwargs):
+        return original_add_to_pool(*args, **kwargs)
+
+    def side_effect_check_modifiability(*args, **kwargs):
+        return original_check_modifiability(*args, **kwargs)
+
+    def side_effect_validate_data(*args, **kwargs):
+        return original_validate_data(*args, **kwargs)
+
+    def side_effect_prepare_data(*args, **kwargs):
+        return orginal_prepare_data(*args, **kwargs)
+
+    patch_add = mocker.patch("RUFAS.input_manager.InputManager._add_to_pool", side_effect=side_effect_add_pool)
+    patch_check = mocker.patch(
+        "RUFAS.input_manager.InputManager._check_modifiability", side_effect=side_effect_check_modifiability
+    )
+    patch_validate = mocker.patch(
+        "RUFAS.input_manager.InputManager._validate_data", side_effect=side_effect_validate_data
+    )
+    patch_prepare = mocker.patch("RUFAS.input_manager.InputManager._prepare_data", side_effect=side_effect_prepare_data)
```

Comment:
```suggestion
    patch_add = mocker.patch("RUFAS.input_manager.InputManager._add_to_pool", wraps=input_manager._add_to_pool)
    patch_check = mocker.patch(
        "RUFAS.input_manager.InputManager._check_modifiability", wraps=input_manager._check_modifiability
    )
    patch_validate = mocker.patch(
        "RUFAS.input_manager.InputManager._validate_data", wraps=input_manager._validate_data
    )
    patch_prepare = mocker.patch("RUFAS.input_manager.InputManager._prepare_data", wraps=input_manager._prepare_data)
```
I suggest using this pattern for the test instead, it does pretty much the same thing but doesn't add any Mypy errors

Reply:
Thanks Ed, that is really helpful.